services/frontend-service/src/ui/AppBar/LocksDrawer.tsx

```diff
@@ -0,0 +1,171 @@
+/*This file is part of kuberpult.
+
+Kuberpult is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Kuberpult is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with kuberpult.  If not, see <http://www.gnu.org/licenses/>.
+
+Copyright 2021 freiheit.com*/
+import React, { useCallback, useEffect } from 'react';
+import Badge from '@material-ui/core/Badge';
+import WarningRoundedIcon from '@material-ui/icons/WarningRounded';
+import ExpandMoreRounded from '@material-ui/icons/ExpandMoreRounded';
+import { Button, Box, Drawer, Grid, Paper, Typography, Divider, List } from '@material-ui/core';
+
+import { GetOverviewResponse, Commit } from '../../api/api';
+import { theme } from '../App/styles';
+
+import { useStyles } from './styles';
+
+interface ILocks {
+    id: string;
+    message: string;
+    commit?: Commit;
+    type?: string;
+    age?: number;
+}
+
+const AllLocks = (props: { locks: ILocks[]; onClick: (event: React.KeyboardEvent | React.MouseEvent) => void }) => {
+    const { locks, onClick } = props;
+    const classes = useStyles();
+
+    if (locks.length === 0) {
+        return (
+            <Typography noWrap variant="h5" component="div" className="name" width="100%" align="center">
+                {'No locks!'}
+            </Typography>
+        );
+    }
+
+    return (
+        <List className={classes.environments} sx={{ width: 'auto' }}>
+            {locks.map((lock) => (
+                <>
+                    <Grid item xs={12} key={lock.id} onClick={onClick}>
+                        <Paper className="environment">
+                            {(!!lock.age && lock.age) > 2 && <WarningRoundedIcon color="error" />}
```

Comment:
This seems like a lot of logic in one line, please extract a variable, e.g.
````
const isLockOld = lock.age... > 2
const warning = isLockOld ? <Warning.../> : null
```
`{warning}`